### ISSUE TITLE
Show bind errors in js files

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#5785]
Bind errors for JS files should be valid errors, we should show them. The following errors i would expect to be flagged:

```ts
let C = "sss";
let C = 0;  // Error: Cannot redeclare block-scoped variable 'C'.

function f() {
    return;
    return;  // Error: Unreachable code detected.
}

function b() {
    "use strict";
    var arguments = 0;  // Error: Invalid use of 'arguments' in strict mode.
}
```

**Comment by kitsonk** [comment]
Isn't the second issue really a linting issue versus a syntax issue?  Also, if you want to trap that as an error, you need to account for `break`, `catch`, `throw` as well as `return`.  I think jshint, jslint and tslint trap all those already.

**Comment by vladima** [comment]
Reachability checks are already in master (#4788), they just were not reported for javascript files.
